src/Data/Argonaut/Decode/Decoders.purs

```diff
@@ -84,61 +86,69 @@ decodeInt = note (TypeMismatch "Integer") <<< fromNumber <=< decodeNumber
 decodeString :: Json -> Either JsonDecodeError String
 decodeString = caseJsonString (Left $ TypeMismatch "String") Right
 
-decodeNonEmpty_Array 
+decodeNonEmptyString :: Json -> Either JsonDecodeError NonEmptyString
+decodeNonEmptyString json =
+  note (Named "NonEmptyString" $ UnexpectedValue json)
+    =<< map (NonEmptyString.fromString) (decodeString json)
+
+decodeNonempty :: Json -> Either JsonDecodeError String
+decodeNonempty = map NonEmptyString.toString <<< decodeNonEmptyString
```

Comment:
I'd rather not include this function; it's pretty easy to do `map NonEmptyString.toString` in your own code and we don't otherwise provide functions beyond the initial parse in the library.